src/workloads/query/SlidingWindows.yml

```diff
@@ -127,6 +127,189 @@ Actors:
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {docs: {
+              $push: "$temp",
+              window: {documents: [-500, 0]}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: FirstSmallWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {first: {
+              $first: "$temp",
+              window: {documents: [-10, 0]}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: FirstLargeWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {first: {
+              $first: "$temp",
+              window: {documents: [-500, 0]}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: LastSmallWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {last: {
+              $last: "$temp",
+              window: {documents: [0, 10]}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: LastLargeWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {last: {
+              $last: "$temp",
+              window: {documents: [0, 500]}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: ShiftSmallWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {shifted: {
+              $shift: {
+                  output: "$temp",
+                  by: 10,
+                  default: "Not available"
+               }}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: ShiftLargeWindow
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{$setWindowFields: {
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {shifted: {
+              $shift: {
+                  output: "$temp",
+                  by: 500,
+                  default: "Not available"
+               }}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: DocumentNumber
```

Comment:
$documentNumber/$rank/$denseRank probably makes more sense in the CumulativeWindows.yml.